### PR TITLE
Circuit executor plugins documentation

### PR DESCRIPTION
**Pull request by buehlefs**
Add documentation for circuit executor plugins.

- Update doc dependencies
- Fix some doc issues
- Add new entry in table of contents for special plugin types / plugin interfaces
- Add documentation for circuit executor plugins

### CODE REVIEW

File: docs/plugin-types/circuit-executor.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    Execution options that are specific to specific pligins should have unique names.
+
+    | Attribute | Description                                      |
+    |:----------|:-------------------------------------------------|
+    | `shots`   | The number of shots to run the circuit.          |
+
+
+Circuit executor plugins can have two types of direct parameter inputs:
+
+ *  **Execution optiones**: circuit executor plugins can expose some of the execution options as parameters.
```
Comment (mhinkie) [inline docs/plugin-types/circuit-executor.md:72]: optiones/options. 
Also: This section is hard to understand - does that mean the direct parameter inputs "execution options" override the file-input-based "execution options"?

**Review by mhinkie — CHANGES_REQUESTED** [review]

File: docs/plugin-types/circuit-executor.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+### Main Interaction
+
+The main input of the plugin is therefore such a circuit along with specific execution options.
+Both must be provided in the starting step of the plugin flow, but the execution options are optional.
+The plugins can make some execution options directly available as parameters for the user.
+However, all these parameters must be optional so that the circuit executors can also be used in more automated settings.
+
+Some circuit executor plugins may require authentication credentials in order to access the actual quantum execution resource.
+Those plugins can expose the authentication form as part of the starting step as optional inputs.
+If the credentials are not provided in the first step, a the plugin should ask for the credentials again in a substep form.
```
Comment (PhilWun) [inline docs/plugin-types/circuit-executor.md:43]: an "a" to much at "..., a the plugin..."

File: docs/plugin-types/circuit-executor.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+ *  **Quantum Circuit** (parameter: `circuit`):\
+    The data input **must** have the data-type [`executable/circuit`](../data-formats/examples/executables.rst#executable-circuit).
+ *  **Execution Options** (parameter `executionOptions`):\
+    The data input **must** be available, but may not receive a value (i.e., this input should be treated as **optional**).
+    The data input **must** have the data-type [`provenance/execution-options`](../data-formats/examples/provenance.rst#provenance-execution-options).
+
+    The execution options **must** support `shots` as the number of shots to simulate/execute.
+    Unknown execution options must be ignored.
+    Execution options that are specific to specific pligins should have unique names.
```
Comment (PhilWun) [inline docs/plugin-types/circuit-executor.md:63]: typo in "plugins"

**Review by PhilWun — CHANGES_REQUESTED** [review]

File: docs/plugin-types/circuit-executor.md (hunk 1 above)
Comment (buehlefs) [reply docs/plugin-types/circuit-executor.md]: I have changed the text to make this more clear

**Review by mhinkie — APPROVED** [review]
looks good

**Pull request merged**